shared/constants/swaps.ts

```diff
@@ -16,60 +16,88 @@ export const CONTRACT_DATA_DISABLED_ERROR = 'contract-data-disabled';
 export const OFFLINE_FOR_MAINTENANCE = 'offline-for-maintenance';
 export const SWAPS_FETCH_ORDER_CONFLICT = 'swaps-fetch-order-conflict';
 
-// An address that the metaswap-api recognizes as the default token for the current network, in place of the token address that ERC-20 tokens have
+// An address that the metaswap-api recognizes as the default token for the current network,
+// in place of the token address that ERC-20 tokens have
 const DEFAULT_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000000';
 
-export const ETH_SWAPS_TOKEN_OBJECT = {
+type SwapsTokenObjectDefinition = {
+  /**
+   * The symbol of token object
+   */
+  symbol: string;
+  /**
+   * The name for the network
+   */
+  name: string;
+  /**
+   * An address that the metaswap-api recognizes as the default token
+   */
+  address: string;
+  /**
+   * Number of digits after decimal point
+   */
+  decimals: number;
+  /**
+   * URL for token icon
+   */
+  iconUrl: string;
+};
+
+export const ETH_SWAPS_TOKEN_OBJECT: SwapsTokenObjectDefinition = {
   symbol: CURRENCY_SYMBOLS.ETH,
   name: 'Ether',
   address: DEFAULT_TOKEN_ADDRESS,
   decimals: 18,
   iconUrl: ETH_TOKEN_IMAGE_URL,
-};
+} as const;
```

Comment:
Thanks for pointing this one out Elliot ! I didn't pay attention earlier. 
But when i'm checking the usage, it is not mutated when used
```
export const ARBITRUM_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
  ...ETH_SWAPS_TOKEN_OBJECT,
} as const;

export const OPTIMISM_SWAPS_TOKEN_OBJECT: SwapsTokenObject = {
  ...ETH_SWAPS_TOKEN_OBJECT,
} as const;
```

I would suggest keeping it this way because if we switch to type we cannot take advantage of predefined const in this object, eg: `DEFAULT_TOKEN_ADDRESS`, `ETH_TOKEN_IMAGE_URL`, 

Let me know your thoughts :)

Reply:
@DDDDDanica the as const only makes sense without the type like @mcmire mentioned. Let me illustrate:

```ts
const SomeObject = {
  x: 'three',
};

const SomeOtherObject = {
  x: 'four',
};

const someFunction(a) {
  return a.x; // inferred as string because SomeObject.x is widened as string
}

someFunction(SomeObject);
someFunction(SomeOtherObject);
```
Once you make an interface that applies to both SomeObject and SomeOtherObject that types the values as literals then you no longer will get the type widening and the as const can be avoided.

Reply:
makes sense ! I'll keep the interface `SwapsTokenObject` and remove `as const` for `ETH_SWAPS_TOKEN_OBJECT`